lib/cocoapods-core/dependency.rb

```diff
@@ -69,11 +90,23 @@ class Dependency
     #
     def initialize(name = nil, *requirements)
       if requirements.last.is_a?(Hash)
-        external_source = requirements.pop.select { |_, v| !v.nil? }
-        @external_source = external_source unless external_source.empty?
-        unless requirements.empty?
-          raise Informative, 'A dependency with an external source may not ' \
-            "specify version requirements (#{name})."
+        additional_params = requirements.pop.select { |_, v| !v.nil? }
+        additional_params = nil if additional_params.empty?
+
+        if additional_params && additional_params[:source]
+          # This dependency specifies the exact source podspec repo to use.
+          @podspec_repo = additional_params[:source]
```

Comment:
this should be `if additional_params && @podspec_repo = additional_params.delete(:source)`